src/util/simplify_expr.cpp

```diff
@@ -2058,26 +2056,33 @@ bool simplify_exprt::simplify_byte_update(byte_update_exprt &expr)
 
     for(const auto &component : components)
     {
-      mp_integer m_offset=
-        member_offset(struct_type, component.get_name(), ns);
-      mp_integer m_size_bits=pointer_offset_bits(component.type(), ns);
-      mp_integer m_size_bytes=m_size_bits/8;
+      auto m_offset = member_offset(struct_type, component.get_name(), ns);
+
+      auto m_size_bits = pointer_offset_bits(component.type(), ns);
+
+      if(!m_offset.has_value() || !m_size_bits.has_value() || *m_size_bits == 0)
+      {
+        result_expr.make_nil();
+        break;
+      }
 
       // can we determine the current offset, and is it a byte-sized
       // member?
-      if(m_offset<0 ||
-         m_size_bits<=0 ||
-         m_size_bits%8!=0)
+      if((*m_size_bits) % 8 != 0)
       {
         result_expr.make_nil();
         break;
       }
+
+      mp_integer m_size_bytes = (*m_size_bits) / 8;
+
       // is that member part of the update?
-      else if(m_offset+m_size_bytes<=offset_int)
+      if(*m_offset + m_size_bytes <= offset_int)
         continue;
       // are we done updating?
-      else if(update_size>0 &&
-              m_offset>=offset_int+update_size)
+      else if(
+        update_size.has_value() && *update_size > 0 &&
+        *m_offset >= offset_int + *update_size)
         break;
```

Comment:
consider putting braces around `break` (preceding multi-line conditional) to improve readability